packages/components/table/src/sortable_table.tsx

```diff
@@ -59,16 +62,19 @@ export const SortableTable = <RecordData extends SortableRecord>({
         return resortedRecords;
       });
     },
-    [setSortedRecords]
+    [onSort, setSortedRecords]
   );
 
   useEffect(() => {
-    if (!onSort) return;
-
-    if (sortedRecords !== records) {
-      onSort(sortedRecords);
+    if (onSortRef.current) {
+      onSortRef.current(sortedRecords);
+      onSortRef.current = undefined;
```

Comment:
Shout out @dgalarza. One of your recent PRs really helped with properly calling the onSort at the right time.